packages/flutter_tools/lib/src/macos/macos_device.dart

```diff
@@ -77,10 +77,18 @@ class MacOSDevice extends Device {
       Cache.releaseLockEarly();
       await buildMacOS(FlutterProject.current(), debuggingOptions?.buildInfo);
     }
+
+    // Ensure that the executable is locatable.
+    final String executable = package.executable(debuggingOptions?.buildInfo?.mode);
+    if (executable == null) {
```

Comment:
Added this check so that the tool exits gracefully, rather than crashing, if the name of the executable couldn't be determined.